jlexer/lexer.go

```diff
@@ -621,9 +633,13 @@ func (r *Lexer) Bool() bool {
 		r.fetchToken()
 	}
 	if !r.Ok() || r.token.kind != tokenBool {
+		if *UseSemanticErrors { // FIXME: remove copypaste from all methods.
```

Comment:
Copy-paste could be avoided this way: first you make errInvalidToken call errSemantic, then you could just leave the errInvalidToken call here. errSemantic can check the flag and fallback to setting a fatal error if semantic errors are disabled.